Unstable `make -j`.

I've noticed that if I `make -j` binder when `cmake` was run with tests enabled, it might sometimes fail with
```
#22 7.131 make[2]: Leaving directory '/build'
#22 7.138 [ 18%] Built target binder
#22 7.140 make[1]: Leaving directory '/build'
#22 7.141 make: *** [Makefile:144: all] Error 2
#22 ERROR: executor failed running [/bin/sh -c make "-j$(nproc)"]: exit code: 2
```

I ran it with `binder` checked out from the current _master_ branch, specifically this commit:  https://github.com/RosettaCommons/binder/commit/7ac60d98ac95ab2d2b47bb46212f54d55f8face0 I have not debugged this or checked other commits. I would recommend using `git bisect` to isolate where the issue came from.

Here is the dockerfile I used:
```docker
FROM ubuntu:20.04 as base

# Config
ARG BRANCH="master"
ARG CLANG_VERSION=14

# General dependencies
ENV DEBIAN_FRONTEND=noninteractive
RUN apt-get update \
 && apt-get install -yq wget gnupg \
 && rm -rf /var/lib/apt/lists/*

# Add llvm repo
RUN wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
RUN echo "deb http://apt.llvm.org/focal/ llvm-toolchain-focal-${CLANG_VERSION} main" >> /etc/apt/sources.list

# Run dependencies
RUN apt-get update \
 && apt-get install -yq "clang-${CLANG_VERSION}" \
 && rm -rf /var/lib/apt/lists/*

FROM base as build

# Build dependencies
RUN apt-get update
RUN apt-get install -yq \
	"libclang-${CLANG_VERSION}-dev" \
	cmake \
	git

# pybind11
RUN git clone https://github.com/pybind/pybind11 /pybind11
RUN apt-get install -yq python3-pip python3-dev
RUN pip3 install pytest
WORKDIR "/py-build"
RUN cmake \
	-DCMAKE_CXX_COMPILER="$(which clang++-"${CLANG_VERSION}")" \
	/pybind11
RUN make "-j$(nproc)"
RUN make install

# Clone binder source
ARG REPO="https://github.com/RosettaCommons/binder.git"
COPY ./binder /binder

# Binder
WORKDIR "/build"
RUN cmake \
	-DCMAKE_CXX_COMPILER="$(which clang++-"${CLANG_VERSION}")" \
	-DBINDER_ENABLE_TEST=ON \
	/binder
RUN make "-j$(nproc)"
```

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -36,6 +36,7 @@ else()
       -- -x c++ -std=c++11   -I . -I ${CMAKE_CURRENT_SOURCE_DIR}  -isystem /  -I ${CLANG_INCLUDE_DIRS}   -iwithsysroot${LibClang_INCLUDE_DIR}
       WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}" VERBATIM)
   endif()
+  add_dependencies(target${stestnamenodot}cpp binder)
 endif()
 endmacro( binder_src  stestname)
 

